samples/fabric/simpleassettransfer/bondasset.go

```diff
@@ -123,49 +140,76 @@ func (s *SmartContract) CreateAsset(ctx contractapi.TransactionContextInterface,
 	return ctx.GetStub().PutState(getBondAssetKey(assetType, id), assetJSON)
 }
 
-// PledgeAsset locks an asset for transfer to a different ledger/network.
-func (s *SmartContract) PledgeAsset(ctx contractapi.TransactionContextInterface, assetType, id, remoteNetworkId, recipientCert string, expiryTimeSecs uint64) error {
-	// Look up the asset details using app-specific-logic
-	assetKey := getBondAssetKey(assetType, id)
-	assetJSON, err := ctx.GetStub().GetState(assetKey)
+// ReadAsset returns the asset stored in the world state with given id.
+// This function is called with the parameter inUpdateOwnerContext value as false, except in the update-owner context
+func (s *SmartContract) ReadAsset(ctx contractapi.TransactionContextInterface, assetType, id string) (*BondAsset, error) {
+	asset, err := getBondAsset(ctx, assetType, id)
 	if err != nil {
-		return fmt.Errorf("failed to read asset record from world state: %v", err)
+		return nil, err
 	}
-	if assetJSON == nil {
-		return fmt.Errorf("the asset %s does not exist", id)
+	if !checkAccessToAsset(s, ctx, asset) {
+		return nil, fmt.Errorf("cannot access Bond Asset %s", id)
 	}
+	return asset, nil
+}
 
-	// Ensure that the client is the owner of the asset using app-specific logic
-	if !checkAccessToAsset(s, ctx, assetJSON) {
-		return fmt.Errorf("Canot pledge asset %s", id)
+// DeleteAsset deletes an given asset from the world state.
+func (s *SmartContract) DeleteAsset(ctx contractapi.TransactionContextInterface, assetType, id string) error {
+	
+	// Read the asset (which internally check access)
+	asset, err := s.ReadAsset(ctx, assetType, id)
+	if err != nil {
+		return err
 	}
+	return ctx.GetStub().DelState(getBondAssetKey(asset.Type, asset.ID))
+}
 
-	// Get asset owner using app-specific-logic
-	var asset BondAsset
-	err = json.Unmarshal(assetJSON, &asset)
+// PledgeAsset locks an asset for transfer to a different ledger/network.
+func (s *SmartContract) PledgeAsset(ctx contractapi.TransactionContextInterface, assetType, id, remoteNetworkId, recipientCert string, expiryTimeSecs uint64) (string, error) {
+	// Check if asset is pledged already
+	bondAssetPledgeMap, err := getAssetPledgeIdMap(ctx, assetType, id)
+	if (err == nil) && (bondAssetPledgeMap.RemoteNetworkID == remoteNetworkId) && (bondAssetPledgeMap.Recipient == recipientCert) {
+		return bondAssetPledgeMap.PledgeID, nil
+	} else if (err == nil) {
+		return bondAssetPledgeMap.PledgeID, fmt.Errorf("asset %s is already pledged with pledgeId %s for different recipient", id, bondAssetPledgeMap.PledgeID)
+	}
+	
```

Comment:
Remove whitespaces.

Reply:
Keeping it here to not make it look cluttered...